### DISCUSSION TITLE
Scroll method in alpine

### DISCUSSION THREAD

**Discussion by Chandler-Zhu** [alpinejs/alpine#1332] [1. Help] [answered]
Is there a method in alpine that controls page scrolling, just like `scrollTop()` or `scrollLeft()` in jQuery ? Or maybe simulating the action of pressing `Home`  key on a keyboard which also scrolls the page to the top

**Comment by gwleuverink** [comment, marked as answer]
There are no scroll helper methods in Alpine. But the situation you sketch is quite achievable with only javascript. 

Consider this code snippet. 

``` html
<div x-data
    x-on:keydown.slash.window="

        // Early return when currently typing in an input
        var inputs = ['input', 'select', 'button', 'textarea'];
        if (document.activeElement && inputs.indexOf(document.activeElement.tagName.toLowerCase()) !== -1) {
            return;
        }

        // Otherwise prevent default & scroll to top
        $event.preventDefault()
        window.scrollTo(0, 0);
    "
>
```

Howerver, I personally feel placing this logic in an alpine component is a bit icky. This adds global page behavior to a random div somewhere in the document. I've done this in the past and am a bit torn on it. Usually I just place stuff like this in a separate plain ol JS file with global event listeners, keeps those things grouped together at least.

**Reply by Chandler-Zhu** [reply]
What would happen if I remove `$event.preventDefault()` ?

**Reply by gwleuverink** [reply]
I just copied this snippet from a project and altered it somewhat so it scrolls. Maybe you don't need the boilerplate in your case.

preventDefault stops the event's default behavior being triggered. In the example I'm listening to the `/` key, which otherwise also triggers the quickfind menu (in my browser at least).

So if you strip it down this could be fine also `x-on:keydown.home.prevent.window="window.scrollTo(0, 0)"` (though I don't have a home key so can't check)

**Reply by Chandler-Zhu** [reply]
Is there a way to scroll to a certain element with vanilla JS ?
